spec/unit/provider/package/windows/registry_uninstall_entry_spec.rb

```diff
@@ -22,4 +24,40 @@
       expect(quiet_uninstall_string_key).to eql "QuietUninstallStringPath"
     end
   end
+
+  describe "when pass 'nil' to find_entries", :windows_only do
```

Comment:
1. Minor change: It's better to use `context` instead of `describe` with `when` statement: https://github.com/reachlocal/rspec-style-guide#context-descriptions.

2. Since `windows_only` filter is applied to all new specs. We can have a common `describe` block for it. eg.
```
describe ".find_entries", :windows_only do
   context "when pass 'nil'" do
     .....
     .....
   end
  
   context "when pass empty string" do
     .....
     .....
   end
.....
end
```

3. If these testcases are actually calling `windows registry`, then they should go in functional testcases. We should be mocking `::Win32::Registry` for unit testcases.